### PR TITLE
Sc 9505/unified env variable management

### PR DESCRIPTION
**Pull request by andriitserkovnyi**
## PR Description

Guides for defining parameters

## Checklist
- [x] I agree with the Code Contribution License Agreement in CONTRIBUTING.md

### CODE REVIEW

File: docs/scos/dev/the-docker-sdk/202204.0/define-parameters-and-secrets-locally.md [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+
+```text
+{SECRET_NAME}='{SECRET_VALUE}'
+{PARAMETER_NAME}=`{PARAMETER_VALUE}`
+```
+
+3. Optional: Check your environment's parameters:
+
+```bash
+list-env
```
Comment (zyuzka) [inline docs/scos/dev/the-docker-sdk/202204.0/define-parameters-and-secrets-locally.md:60]: docker/sdk list-env

File: docs/scos/dev/the-docker-sdk/202204.0/define-parameters-and-secrets-locally.md [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+```
+    This generates the `.env.docker.local` file and adds it to `.gitignore`.
+
+2. In `.env.docker.local`, define parameter and secret values:
+
+```text
+{SECRET_NAME}='{SECRET_VALUE}'
+{PARAMETER_NAME}=`{PARAMETER_VALUE}`
+```
+
```
Comment (zyuzka) [inline docs/scos/dev/the-docker-sdk/202204.0/define-parameters-and-secrets-locally.md:56]: after params definition, the user should run `docker/sdk boot {DEPLOY_FILE_NAME}`

File: docs/cloud/dev/spryker-cloud-commerce-os/define-parameters-and-secrets-in-sccos-environments.md [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+
+To define parameters and their values, do the following:
+
+1. To define parameters locally, follow [Define parameters and secrets locally](/docs/scos/dev/the-docker-sdk/{{site.version}}/define-parameters-and-secrets-locally.html).
+2. Push the updates to the SCCCOS environment.
+3. Deploy the application with the updated configuration by following one of the following docs:
+    * [Deploying in a staging environment](/docs/cloud/dev/spryker-cloud-commerce-os/deploying-in-a-staging-environment.html)
+    * [Deploying in a production environment](/docs/cloud/dev/spryker-cloud-commerce-os/deploying-in-a-production-environment.html)
+        This should update the parameters in the [Parameter Store](https://eu-central-1.console.aws.amazon.com/systems-manager/parameters).
+4. To define parameter values, fill out the [parameter change form](https://spryker.force.com/support/s/hosting-change-requests/change-request-change-parameter).
```
Comment (hatemzidi-pro) [inline docs/cloud/dev/spryker-cloud-commerce-os/define-parameters-and-secrets-in-sccos-environments.md:24]: they don't need to go through support to change the value. they can do it, for now, from the AWS Param Store UI

File: docs/scos/dev/the-docker-sdk/202204.0/define-parameters-and-secrets-locally.md [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+        - name: {PARAMETER_NAME}
+          bucket: {BUCKET}
+          default: {DEFAULT_VALUE}
+          grant: {PERMISSIONS}
+...
+```
+| PARAMETER | DESCRIPTION |
+| - | - |
+| name | Unique identifier of a secret or parameter. You will use it when defining its value in `.env.docker.local`. |
+| grant | IAM users' permissions. `limited` provides read while `public` provides read and write permissions. The default one is `limited`. |
```
Comment (hatemzidi-pro) [inline docs/scos/dev/the-docker-sdk/202204.0/define-parameters-and-secrets-locally.md:31]: i guess "IAM" is irrelevant here. 
it's just permission from the customer's prospective.

File: docs/cloud/dev/spryker-cloud-commerce-os/managing-parameters-in-the-parameter-store.md [hunk 5]
```diff
@@ -1,20 +0,0 @@
...
-  - /docs/en/managing-parameters-in-the-parameter-store
----
-
-This document describes how you can add new parameters and edit the values of existing parameters. You can use parameters for multiple purposes, like storing mail server details or providing Composer authentication details to the build and deploy process securely.
-
-## Adding new parameters
-
-Currently, only our team can add new parameters. To add a new parameter, fill out the [parameter creation form](https://spryker.force.com/support/s/hosting-change-requests/change-request-new-param-store-variable).
-
-## Changing parameters
```
Comment (hatemzidi-pro) [inline docs/cloud/dev/spryker-cloud-commerce-os/managing-parameters-in-the-parameter-store.md:18]: this part is still relevant. Except we will give them (customers) a "strict" access to SSM.

File: docs/scos/dev/the-docker-sdk/202204.0/deploy-file/deploy-file-reference-1.0.md [hunk 6]
```diff
@@ -136,6 +136,26 @@ environment: 'docker'
 
 ***
 
+### environment-configuration: secrets:
+
+Defines secrets and their configuration:
+
+* `environment-configuration: secrets: name:`—Secret's name.
+* `environment-configuration: secrets: grant:`—IAM users' permissions. `limited` provides read while `public` provides read and write permissions. The default one is `limited`.
```
Comment (hatemzidi-pro) [inline docs/scos/dev/the-docker-sdk/202204.0/deploy-file/deploy-file-reference-1.0.md:144]: the word "IAM" is to omit

File: docs/cloud/dev/spryker-cloud-commerce-os/define-parameters-and-secrets-in-sccos-environments.md (hunk 3 above)
Comment (andriitserkovnyi) [reply docs/cloud/dev/spryker-cloud-commerce-os/define-parameters-and-secrets-in-sccos-environments.md:24]: While writing this doc, I've been told that customers don't have perms to do that. Is that something we've changed recently?

File: docs/scos/dev/the-docker-sdk/202204.0/deploy-file/deploy-file-reference-1.0.md [hunk 7]
```diff
@@ -136,6 +136,26 @@ environment: 'docker'
...
 
+### environment-configuration: secrets:
+
+Defines secrets and their configuration:
+
+* `environment-configuration: secrets: name:`—Secret's name.
+* `environment-configuration: secrets: grant:`—Users' permissions. `limited` provides read while `public` provides read and write permissions. The default one is `limited`.
+* `environment-configuration: secrets: bucket:`—Defines what the secret is used for. Acceptable values are `app`, `scheduler`, `pipeline`, and `common`. `common` is the default value. Common secrets can be used by all the buckets.
+
+### environment-configuration: params:
```
Comment (hatemzidi-pro) [inline docs/scos/dev/the-docker-sdk/202204.0/deploy-file/deploy-file-reference-1.0.md:147]: same as above, this is not to publish for the current release.

File: _drafts/unified-variable-management-sc-9505/define-parameters-and-secrets-in-sccos-environments.md [hunk 8]
```diff
@@ -0,0 +1,36 @@
...
+To define parameters and their values, do the following.
+
+## Prerequisites
+
+1. [Define parameters and secrets](/docs/scos/dev/the-docker-sdk/{{site.version}}/define-parameters-and-secrets.html).
+2. Push the updates to the SCCOS environment.
+3. Deploy the application with the updated configuration by following one of the following docs:
+    * [Deploying in a staging environment](/docs/cloud/dev/spryker-cloud-commerce-os/deploying-in-a-staging-environment.html)
+    * [Deploying in a production environment](/docs/cloud/dev/spryker-cloud-commerce-os/deploying-in-a-production-environment.html)
+        This should update the parameters in the Parameter Store.
```
Comment (VadymSachenko) [inline _drafts/unified-variable-management-sc-9505/define-parameters-and-secrets-in-sccos-environments.md:25]:
```suggestion
        This updates the parameters in the Parameter Store.
```

File: _drafts/unified-variable-management-sc-9505/define-parameters-and-secrets-in-sccos-environments.md [hunk 9]
```diff
@@ -0,0 +1,36 @@
...
+
+## Define parameter and secret values         
+
+1. In the AWS Management Console, go to **Services** > **Parameter Store**.
```
Comment (VadymSachenko) [inline _drafts/unified-variable-management-sc-9505/define-parameters-and-secrets-in-sccos-environments.md:29]:
```suggestion
1. In the AWS Management Console, go to **Services > Parameter Store**.
```

File: _drafts/unified-variable-management-sc-9505/define-parameters-and-secrets.md [hunk 10]
```diff
@@ -0,0 +1,67 @@
+---
+title: Define parameters and secrets
+description: Learn how to declare parameters and define their values.
+last_updated: July 28, 2022
```
Comment (VadymSachenko) [inline _drafts/unified-variable-management-sc-9505/define-parameters-and-secrets.md:4]:
```suggestion
last_updated: Dec 2, 2022
```

File: _drafts/unified-variable-management-sc-9505/define-parameters-and-secrets.md [hunk 11]
```diff
@@ -0,0 +1,67 @@
...
+```
+
+## Define parameter and secret values locally
+
+1. Generate a file for parameter and secret values:
+
+```bash
+docker/sdk generate-env
+```
+    This generates the `.env.docker.local` file and adds it to `.gitignore`.
```
Comment (VadymSachenko) [inline _drafts/unified-variable-management-sc-9505/define-parameters-and-secrets.md:48]:
```suggestion

This generates the `.env.docker.local` file and adds it to `.gitignore`.
```

File: _drafts/unified-variable-management-sc-9505/deploy-file-reference-1.0.md [hunk 12]
```diff
@@ -0,0 +1,20 @@
+...
+
+### environment-configuration: secrets:
+
+Defines secrets and their configuration:
+
+* `environment-configuration: secrets: name:`—Secret's name.
```
Comment (VadymSachenko) [inline _drafts/unified-variable-management-sc-9505/deploy-file-reference-1.0.md:7]: Why are phrases after an em-dash uppercase?

**Review by VadymSachenko — CHANGES_REQUESTED** [review]

**Review by VadymSachenko — APPROVED** [review]

**Pull request merged**